### PR TITLE
[JS] Other: Functions

### LINKED ISSUE #247: JS: Functions
# Functions
Functions in JavaScript

## Directory
```
Topics/5.Syntax/6.Functions
```

## Description
- What is scope?
- What is the lifetime of JavaScript variables?
- Basic function syntax.
- What are parameters in functions?
- Types of functions.
- Why use functions?

### PR DESCRIPTION
**Pull request by sushovanb02**
<hr>

## Description 📜

Created a file containing extra information about functions.
Fixes #247 

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] JavaScript

<hr>
 
## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by sushovanb02** [comment]
@paramsiddharth sir, As you said I did my work!!

**Review by paramsiddharth — APPROVED** [review]
@Creatoon @dakshsethi @s4-2 Please review further.

**Comment by sushovanb02** [comment]
@paramsiddharth can you add the javascript and gwoc label, i don't know how to do it in this one

**Review by Creatoon — APPROVED** [review]

**Comment by paramsiddharth** [comment]
@dakshsethi @s4-2 Please review.

File: JavaScript/Topics/5.Syntax/6.Functions/README.md [hunk 1]
```diff
@@ -16,12 +22,54 @@ This time javascript throws an error. Why is it so?
...
+    - Variables are accessible only inside block (block scoped)
+    - HOWEVER, this only applies to let and const variables!
+    - Functions are also block scoped (only in strict mode)
+    ```js
+    if (year >= 1981 && year <= 1996){
+        const millenial = true;
+        const food = 'Pasta';
+    }
+    console.log(millenial) //Reference Error
+    ```
```
[the remark this answers is not in the document]
Comment (sushovanb02) [reply JavaScript/Topics/5.Syntax/6.Functions/README.md]: @s4-2 done as u said

**Review by s4-2 — APPROVED** [review]

**Comment by s4-2** [comment]
Looks good to me, well done:)

**Comment by sushovanb02** [comment]
@paramsiddharth All mentors have reviewed, I think it's good to be merged

**Pull request merged**